### PR TITLE
Add optional health query parameter to push notification to resolve #961

### LINKED ISSUE #961: allow push monitoring set status via query param (in addition to "upside down" checkbox)
### ⚠️ Please verify that this feature request has NOT been suggested before.

- [X] I checked and didn't find similar feature request

### 🏷️ Feature Request Type

API

### 🔖 Feature description

Allow `push` webhook to set the health via param.

I would like to propose and see if there are interest in allowing `push` monitors to set the health of the monitor. 

### ✔️ Solution

Proposal: for `push` webhook, optionally allow a `set_health` param. If set, expects : `true | false | 1 | 0`. If set, will set the health, ignoring the default upside down setting. 

### ❓ Alternatives

(I can't think of any. It is functional as it is, but just a bit of paper cut. UK already provides 99% of what we would like in a OS monitoring tool.)

### 📝 Additional Context

My use case:

I have a few github actions running on a schedule. They do different tasks: some monitor API, some carry out simple e2e tasks (too complex for UK HTTP with keywords monitor). At the conclusion, if they succeed, they send heart beat to UK `push` monitor.

The `push` monitor's heartbeat intervals are set at 30 minutes (cron jobs is scheduled once every 10 minute). I suspect this sometimes cause false positives (github action does not start on time, job takes too long, etc). 

I will be happy to try to provide a PR, if there are any interest?

### PR DESCRIPTION
**Pull request by corsmond**
# Add optional health query parameter to push notification to allow setting health UP or DOWN

Fixes #961

## Type of change

Please delete options that are not relevant.

- New feature (non-breaking change which adds functionality)
- This change requires a documentation update

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I ran ESLint and other linters for modified files
- [x] I have performed a self-review of my own code and test it
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

## Screenshots (if any)

Please do not use any external image service. Instead, just paste in or drag and drop the image here, and it will be uploaded automatically.

### CODE REVIEW

**Comment by trogper** [comment]
I have checked your commit and it changes almost every line in the file, therefore it is not very clear which lines are changed and how.
I would like to object to "My code follows the style guidelines of this project".  Your commit changes double-quotes to single-quotes and replaces spaces with tabs. Good practice is to view changes you made before commiting and to commit only the lines which you modified/where you changed the logic.
Please run eslint and read the output. I know there are a lot of other files listed, but you can skip these and read only your own.

Other than that I would suggest to use `status` instead of `health`.
- UK uses word `status` and not `health`
- Health implies a range of values (probably float), whereas status implies only yes/no (boolean).

I was actually implementing the change myself, but then I remembered to check, whether there is already pull request 🙂.

**Comment by trogper** [comment]
do not forget to add the parameter to `EditMonitor.vue` to `pushURL()` function and `<!-- Push URL -->` section

**Comment by trogper** [comment]
thanks, that's much better. However there are still small changes to lines where there is no logic modified (added/removed spaces, split/merged lines).

**Comment by chakflying** [comment]
So this PR really only changes 3 lines? I think it would get accepted faster if you only leave those in.

**Comment by trogper** [comment]
yes, basically it is only 5-6 lines (two lines not yet implemented are in `EditMonitor.vue`, see my [fork/branch](https://github.com/trogper/uptime-kuma/commit/3d002b3ce95d51ead40b71c8c31cbe0bb3522e91)

File: server/routers/api-router.js [hunk 1]
```diff
@@ -1,242 +1,240 @@
...
 let router = express.Router();
 
 let cache = apicache.middleware;
 let io = server.io;
 
-router.get("/api/entry-page", async (_, response) => {
-    allowDevAllOrigin(response);
-    response.json(server.entryPage);
+router.get('/api/entry-page', async (_, response) => {
+	allowDevAllOrigin(response);
```
Comment (louislam) [inline server/routers/api-router.js]: It becomes a real tab character. Feel like you are not using our repo's eslint config.

**Comment by corsmond** [comment]
Yes, basically only those 3 lines. Adding DOWN to the imports, checking for health parameter and using that for status. If there is another PR that has it in, you can close mine. I used the .eslintrc.js in the project. Running VSCODE and even disabled prettier, so not sure why it keeps showing other changes.

**Pull request closed without merging**